src/main/java/com/gmail/woosay333/onlinebookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.gmail.woosay333.onlinebookstore.repository.impl;
+
+import com.gmail.woosay333.onlinebookstore.entity.Book;
+import com.gmail.woosay333.onlinebookstore.exception.DataProcessingException;
+import com.gmail.woosay333.onlinebookstore.repository.BookRepository;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```

Comment:
this is redundant, when you have a single constructor
```suggestion
    private final SessionFactory sessionFactory;

```